static/script/devices/mediaplayer/samsung_streaming.js

```diff
@@ -465,49 +469,51 @@ require.def(
-                var range = this._playerPlugin.Execute("GetPlayingRange").split('-');
-                this._range = {
-                    start: range[0],
-                    end: range[1]
-                };
-
-                this.logger.info('Current Duration ' + this._range.end * 1000 + 'ms');
-
-                // this.logger.info("_onMetadata range = {0, "  + this._range.end);
-                // this.logger.info("this._currentTime = " + this._currentTime);       //<TODO> so currentTime is currently in milliseconds relative to midnight! and not to hls window
-
-                if (this._waitingToPlayFrom()) {
-                    this._deferredPlayFrom();
-                }
-
-                this._onFinishedBuffering();
-            },
-
-            _waitingToPlayFrom: function() {
-                return this._targetSeekTime !== undefined;
-            },
-
-            _deferredPlayFrom: function() {
-                this._seekTo(this._targetSeekTime);
-
-                if (this._postBufferingState === MediaPlayer.STATE.PAUSED) {
-                    this._tryPauseWithStateTransition();
+            
+            _updateRange: function () {
+                var self = this;
+                if (this._currentPlayer === this.PlayerEmps.StreamingPlayer) {
+                    var range = this._playerPlugin.Execute("GetPlayingRange").split('-');
+                    this._range = {
+                        start: Math.floor(range[0]),
+                        end: Math.floor(range[1])
+                    };
+                    //don't call range for the next 8 seconds
+                    this._updatingTime = true;
+                    setTimeout(function () {
+                        self._updatingTime = false;
+                    }, self.RANGE_UPDATE_TOLERANCE * 1000);
+                } else if (this._currentPlayer === this.PlayerEmps.Player) {
+                    var duration = this._playerPlugin.Execute("GetDuration")/1000;
+                    this._range = {
+                        start: 0,
+                        end: duration
+                    }
                 }
-                this._targetSeekTime = undefined;
             },
 
 
             _onCurrentTime: function(timeInMillis) {
 
-                //this.logger.info("_onCurrentTime "  + timeInMillis);
+                //this.logger.info("_onCurrentTime: "  + timeInMillis);
 
                 this._currentTime = timeInMillis / 1000;
                 this._onStatus();
                 this._currentTimeKnown = true;
 
+                //[optimisation] do not call player API periodically in HLS live
+                // - calculate range manually when possible
+                // - do not calculate range if player API was called less than RANGE_UPDATE_TOLERANCE seconds ago
+                if (this._isLiveMedia() && this.isLiveRangeOutdated()) {
+                    this._range.start += 8;
```

Comment:
Actually this is exactly the reason why in _getClampedTimeForPlayFrom range is "synchronised" with platform.
If you prefer to get it more often from device then I would suggest to leave current state "as is" and rather add extra periodical check for example every 10~20 minutes to ensure that we still display correct range.